python/cuml/common/logger.pyx

```diff
@@ -73,6 +77,15 @@ level_critical = CUML_LEVEL_CRITICAL
 """Disables all log messages"""
 level_off = CUML_LEVEL_OFF
 
+cdef void _log_callback(int lvl, const char * msg) with gil:
+    print(msg.decode('utf-8'), end='')
+
+cdef void _log_flush() with gil:
+    sys.stdout.flush()
```

Comment:
Out of curiosity, why print(end='') instead of sys.stdout.write which would look more parallel with sys.stdout.flush?

Reply:
The [implementation of print](https://github.com/python/cpython/blob/3.7/Python/bltinmodule.c#L1851) includes a bunch of checks that handle edge cases that we'd otherwise have to worry about with `sys.stdout.write`, but now that you mention it, one of those edge cases (`sys.stdout` being `None`) is relevant to `sys.stdout.flush` as well. I'll add a check to the `_log_flush` method.